test/hotspot/jtreg/compiler/vectorapi/TestReverseByteTransforms.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import java.util.Random;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorSpecies;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.ShortVector;
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8287794
+ * @summary Test various reverse bytes ideal transforms.
+ * @requires vm.compiler2.enabled
+ * @modules jdk.incubator.vector
+ * @library /test/lib /
+ * @run driver compiler.vectorapi.TestReverseByteTransforms
+ */
+
+public class TestReverseByteTransforms {
+    static final VectorSpecies<Long> LSPECIES = LongVector.SPECIES_MAX;
+    static final VectorSpecies<Integer> ISPECIES = IntVector.SPECIES_MAX;
+    static final VectorSpecies<Short> SSPECIES = ShortVector.SPECIES_MAX;
+
+    static final int SIZE = 1024;
+    static final int ITERS = 50000;
+
+    static long [] lout = new long[SIZE];
+    static long [] linp = new long[SIZE];
+
+    static int [] iout = new int[SIZE];
+    static int [] iinp = new int[SIZE];
+
+    static short [] sout = new short[SIZE];
+    static short [] sinp = new short[SIZE];
+
+    static void init() {
+        Random r = new Random(1024);
+        for(int i = 0; i < SIZE; i++) {
+            linp[i] = r.nextLong();
+            iinp[i] = r.nextInt();
+            sinp[i] = (short)r.nextInt();
+        }
+    }
+
+    public static void main(String args[]) {
+        init();
+        TestFramework.runWithFlags("-XX:-TieredCompilation", "--add-modules=jdk.incubator.vector");
+        System.out.println("PASSED");
+    }
+
+    @Test
+    @IR(applyIfCPUFeature={"sve", "true"}, failOn = {"ReverseBytesV" , " > 0 "})
```

Comment:
Use " failOn = "ReverseBytesV" " instead?

Reply:
Thanks for noticing this, semantically, failOn NODE is same as failOn NODE , "> 0", tried and tested it, but it looks like a non-standard usage of the option.